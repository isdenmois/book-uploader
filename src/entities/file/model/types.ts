export type UPLOAD_STATE = 'SCAN' | 'IDLE' | 'UPLOAD' | 'FINISH' | 'HAS_ERRORS'

export interface FileData {
  id: string
  path: string
  ext: string
  title: string

  isParsed?: boolean
  isUploaded?: boolean

  author?: string
  progress?: number
  error?: string
}
