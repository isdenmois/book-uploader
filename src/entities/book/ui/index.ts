export * from './book-item'
export * from './book-list'
