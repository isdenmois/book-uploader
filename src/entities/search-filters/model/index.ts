export * from './actions'
export * from './effects'
export * from './events'
export * from './selectors'
export * from './stores'
export * from './types'
