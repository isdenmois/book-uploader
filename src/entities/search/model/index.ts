export * from './actions'
export * from './effects'
export * from './events'
export * from './stores'
export * from './types'
