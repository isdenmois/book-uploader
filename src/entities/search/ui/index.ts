export * from './search-filters'
export * from './book-list'
