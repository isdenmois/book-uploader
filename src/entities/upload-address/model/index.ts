export * from './actions'
export * from './events'
export * from './stores'
