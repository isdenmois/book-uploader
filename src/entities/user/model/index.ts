export * from './actions'
export * from './effects'
export * from './events'
export * from './store'
