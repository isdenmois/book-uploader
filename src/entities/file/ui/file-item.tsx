import React, { FC } from 'react'
import { TouchableOpacity } from 'react-native'

import { FileIcon, Item, ProgressBar, TimesIcon } from 'shared/ui'
import { useConfirm } from 'shared/utils'
import type { FileData } from '../model'

interface Props {
  file: FileData
  disabled?: boolean
  onRemove: (file: FileData) => void
  onParse: (file: FileData) => void
  onShare: (file: FileData) => void
}

export const FileItem: FC<Props> = ({ file, disabled, onRemove, onParse, onShare }) => {
  const parseDisabled = disabled || file.isParsed
  const removeFile = useConfirm('Remove file?', file.title || file.id, () => onRemove(file))

  return (
    <Item
      mb={3}
      onPress={() => onShare(file)}
      onLongPress={parseDisabled ? null : () => onParse(file)}
      disabled={disabled}
      progress={!file.error && <ProgressBar progress={file.progress} color='uploadSelected' />}
      testID={'file-item-' + file.id}
    >
      <FileIcon size={34} color='uploadSelected' text={file.ext} />

      <Item.Text suptitle={file.author} title={file.title} error={file.error} />

      {!disabled && (
        <TouchableOpacity onPress={removeFile} testID='removeFile'>
          <TimesIcon />
        </TouchableOpacity>
      )}
    </Item>
  )
}
