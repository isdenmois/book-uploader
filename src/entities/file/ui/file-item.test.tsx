import React from 'react'
import { act, fireEvent, render, RenderAPI } from '@testing-library/react-native'
import { Alert } from 'react-native'
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
import { mock } from 'shared/utils/test-utils/async'
import { ThemeProvider } from 'shared/ui'
import { FileData } from '../model'
import { FileItem } from './file-item'

describe('FileItem', () => {
  const file: FileData = { id: '123', title: 'Hello', author: 'Somebody', ext: 'epub', path: '/tmp/smbdy-hello.epub' }
  const alert = mock(Alert, 'alert')
  const onRemove = jest.fn()
  const onParse = jest.fn()
  const onShare = jest.fn()

  let r: RenderAPI

  beforeEach(() => {
    r = render(<FileItem file={file} onRemove={onRemove} onParse={onParse} onShare={onShare} />, {
      wrapper: ThemeProvider,
    })
  })

  afterEach(() =>
    act(async () => {
      r.unmount()
    }),
  )

  it('should render', () => {
    expect(r.toJSON()).toMatchSnapshot()
  })

  it('should be able to remove file', () => {
    fireEvent.press(r.getByTestId('removeFile'))

    expect(onRemove).not.toHaveBeenCalled()
    expect(alert).toHaveBeenCalled()

    alert.mock.calls[0][2][1].onPress()

    expect(onRemove).toHaveBeenCalledWith(file)
  })

  it('should call onShare on tap', () => {
    fireEvent.press(r.getByTestId('file-item-123'))

    expect(onShare).toHaveBeenCalledWith(file)
  })

  it('should call onParse on long tap', () => {
    fireEvent(r.getByTestId('file-item-123'), 'longPress')

    expect(onParse).toHaveBeenCalledWith(file)
  })

  it('should show an error', () => {
    r.rerender(
      <FileItem
        file={{ ...file, error: 'Something went wrong' }}
        onRemove={onRemove}
        onParse={onParse}
        onShare={onShare}
      />,
    )
    expect(r.toJSON()).toMatchSnapshot()
  })
})
