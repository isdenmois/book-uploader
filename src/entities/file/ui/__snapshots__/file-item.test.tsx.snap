// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileItem should render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "display": "flex",
      "marginBottom": 24,
      "opacity": 1,
    }
  }
  testID="file-item-123"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={50}
      bbWidth={40}
      focusable={false}
      height={50}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 50,
            "width": 40,
          },
        ]
      }
      vbHeight={50}
      vbWidth={40}
      width={40}
    >
      <RNSVGGroup>
        <RNSVGGroup>
          <RNSVGPath
            d="M24 0H2.4A2.394 2.394 90 000 2.388V48.8a2.394 2.394 90 002.388 2.4H36a2.394 2.394 90 002.4-2.388V14.4H25.6a1.6 1.6 90 01-1.6-1.6zm14.4 12.19v.61H27.2a1.6 1.6 90 01-1.6-1.6V0h.61a2.4 2.4 90 011.7.7l9.79 9.8a2.39 2.39 90 01.7 1.69z"
            fill={4285026800}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
        <RNSVGText
          content={null}
          dx={Array []}
          dy={Array []}
          fill={4294967295}
          font={
            Object {
              "fontSize": "12",
              "fontWeight": "bold",
              "textAnchor": "middle",
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          rotate={Array []}
          x={
            Array [
              "19",
            ]
          }
          y={
            Array [
              "44",
            ]
          }
        >
          <RNSVGTSpan
            content="EPUB"
            dx={Array []}
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={Array []}
            y={Array []}
          />
        </RNSVGText>
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flex": 1,
            "marginLeft": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#787D87",
              "display": "flex",
              "fontSize": 12,
              "lineHeight": 14,
            },
          ]
        }
      >
        Somebody
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#092D5D",
              "display": "flex",
              "fontSize": 16,
            },
          ]
        }
      >
        Hello
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="removeFile"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={512}
        vbWidth={320}
        width={24}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
            fill={4287402915}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "height": 12,
        },
      ]
    }
  />
</View>
`;

exports[`FileItem should show an error 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "display": "flex",
      "marginBottom": 24,
      "opacity": 1,
    }
  }
  testID="file-item-123"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={50}
      bbWidth={40}
      focusable={false}
      height={50}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 50,
            "width": 40,
          },
        ]
      }
      vbHeight={50}
      vbWidth={40}
      width={40}
    >
      <RNSVGGroup>
        <RNSVGGroup>
          <RNSVGPath
            d="M24 0H2.4A2.394 2.394 90 000 2.388V48.8a2.394 2.394 90 002.388 2.4H36a2.394 2.394 90 002.4-2.388V14.4H25.6a1.6 1.6 90 01-1.6-1.6zm14.4 12.19v.61H27.2a1.6 1.6 90 01-1.6-1.6V0h.61a2.4 2.4 90 011.7.7l9.79 9.8a2.39 2.39 90 01.7 1.69z"
            fill={4285026800}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
        <RNSVGText
          content={null}
          dx={Array []}
          dy={Array []}
          fill={4294967295}
          font={
            Object {
              "fontSize": "12",
              "fontWeight": "bold",
              "textAnchor": "middle",
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          rotate={Array []}
          x={
            Array [
              "19",
            ]
          }
          y={
            Array [
              "44",
            ]
          }
        >
          <RNSVGTSpan
            content="EPUB"
            dx={Array []}
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={Array []}
            y={Array []}
          />
        </RNSVGText>
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flex": 1,
            "marginLeft": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#092D5D",
              "display": "flex",
              "fontSize": 16,
            },
          ]
        }
      >
        Hello
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#FD4463",
              "display": "flex",
              "fontSize": 12,
              "lineHeight": 14,
            },
          ]
        }
      >
        Something went wrong
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="removeFile"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={512}
        vbWidth={320}
        width={24}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
            fill={4287402915}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
